db/import_column_family_job.cc

```diff
@@ -262,37 +266,67 @@ Status ImportColumnFamilyJob::GetIngestedFileInfo(
   // Get number of entries in table
   file_to_import->num_entries = props->num_entries;
 
-  ParsedInternalKey key;
-  ReadOptions ro;
-  // During reading the external file we can cache blocks that we read into
-  // the block cache, if we later change the global seqno of this file, we will
-  // have block in cache that will include keys with wrong seqno.
-  // We need to disable fill_cache so that we read from the file without
-  // updating the block cache.
-  ro.fill_cache = false;
-  std::unique_ptr<InternalIterator> iter(table_reader->NewIterator(
-      ro, sv->mutable_cf_options.prefix_extractor.get(), /*arena=*/nullptr,
-      /*skip_filters=*/false, TableReaderCaller::kExternalSSTIngestion));
-
-  // Get first (smallest) key from file
-  iter->SeekToFirst();
-  Status pik_status =
-      ParseInternalKey(iter->key(), &key, db_options_.allow_data_in_errors);
-  if (!pik_status.ok()) {
-    return Status::Corruption("Corrupted Key in external file. ",
-                              pik_status.getState());
-  }
-  file_to_import->smallest_internal_key.SetFrom(key);
-
-  // Get last (largest) key from file
-  iter->SeekToLast();
-  pik_status =
-      ParseInternalKey(iter->key(), &key, db_options_.allow_data_in_errors);
-  if (!pik_status.ok()) {
-    return Status::Corruption("Corrupted Key in external file. ",
-                              pik_status.getState());
+  // If the importing files were exported with Checkpoint::ExportColumnFamily(),
+  // we cannot simply recompute smallest and largest used to truncate range
+  // tombstones from file content, and we expect smallest and largest populated
+  // in file_meta.
+  if (file_meta.smallest.empty()) {
+    assert(file_meta.largest.empty());
+    ReadOptions ro;
+    // During reading the external file we can cache blocks that we read into
+    // the block cache, if we later change the global seqno of this file, we
+    // will have block in cache that will include keys with wrong seqno. We need
+    // to disable fill_cache so that we read from the file without updating the
+    // block cache.
```

Comment:
I think we fixed this years ago in #6429

Reply:
I copy pasted from external_sst_file_ingestion_job.cc. Updated both places.